src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,98 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
+import { pattern } from '../Service/RegularFunction';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+const initialMovieState: Movie = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [movie, setMovie] = useState<Movie>(initialMovieState);
+  const handleMovieChange = (event: React.ChangeEvent<HTMLInputElement>,
+    key: string) => {
+    setMovie(prevMovie => ({ ...prevMovie, [key]: event.target.value }));
+  };
```

Comment:
In the 'handleMovieChange' function, the 'key' argument is of type string. This can be risky because if the key does not exist in the movie object, it will create an error. To prevent this, you should use `keyof Movie` for the key parameter.

```suggestion
  const handleMovieChange = (event: React.ChangeEvent<HTMLInputElement>, key: keyof Movie) => {
    setMovie(prevMovie => ({ ...prevMovie, [key]: event.target.value }));
  };
```